File: 69_Sqrt(x)/solution.c
int mySqrt(int x) {
    if (x == 0) return 0;

    int low = 1, high = x;
    while (low <= high) {
        int mid = low + (high - low) / 2;
        long long sqrt = (long long)mid * mid;

        if (sqrt == x) {
            return mid;
        } else if (sqrt < x) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return high;
}

File: 69_Sqrt(x)/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0

        low, high = 1, x

        while low <= high:
            mid = low + (high - low) // 2
            sqrt = mid * mid

            if sqrt == x:
                return mid
            elif sqrt < x:
                low = mid + 1
            else:
                high = mid - 1

        return high
